js/id/ui/alert.js

```diff
@@ -21,29 +20,67 @@ iD.ui.Alert = function(message,type,stack) {
     if(type === 'success'){alertDiv.classed('green',true);}
 
     function modalWindow() {
+        d3.select('#detailDiv').remove();
         var modalbg = d3.select('body').append('div')
             .attr('id','detailDiv')
             .classed('fill-darken3 pin-top pin-left pin-bottom pin-right', true)
-            .on('click', function() {
+            .on('click',function() {
                 if (d3.event.target === d3.select('#detailDiv').node()) {
```

Comment:
Pretty sure I added the if statement because the window was closing when you clicked both on it and outside of it, instead of just outside of it, as intended. I'll remove it and test, you may be right.